### DISCUSSION TITLE
New to vike, can't understand how to define meta properties

### DISCUSSION THREAD

**Discussion by orangecoloured** [vikejs/vike#1360] [Discussion] [answered]
Trying to build a project using `preact` and it uses `vike`. One of the first things I tried to do is to add custom title and description to a page, following the docs. But I'm getting this error
```
/pages/index/+config.h.ts defines an unknown config title, you need to define the config title by using config.meta https://vike.dev/meta
```

Here's my `+config.h.ts`
```
export default {
  title: 'New title',
}
```

On the `https://vike.dev/meta` I don't see an example of defining a new config property.

**Comment by phonzammi** [comment, marked as answer]
Did you have this file?

```ts
// /pages/+config.h.ts or /renderer/config.h.ts

import type { Config } from 'vike/types'

export default {
  meta: {
    title: {
      // Make the value of `title` available on both the server- and client-side
      env: { server: true, client: true }
    },
    description: {
      // Make the value of `description` available only on the server-side
      env: { server: true }
    }
  }
} satisfies Config

```

**Reply by orangecoloured** [reply]
Alright, thanks. Now it stopped throwing the error and seems like it's passing the title and description. But there's another issue.

Here's part of my `+onRenderHtml.tsx` content:
```
async function render(pageContext) {
	const { documentProps } = pageContext.exports;
	const title = (documentProps && documentProps.title) || 'Vite SSR + Preact';
	const desc =
		(documentProps && documentProps.description) || 'Preact app with Vite and vike';
```

but the title and description are right under `pageContext.exports`. Does it mean that I need to update my `/pages/+config.h.ts` to be like this?
```
import type { Config } from 'vike/types'

export default {
  meta: {
    documentProps: {
      title: {
        // Make the value of `title` available on both the server- and client-side
        env: { server: true, client: true }
      },
      description: {
        // Make the value of `description` available only on the server-side
        env: { server: true }
      }
    }
  }
} satisfies Config
```

**Reply by phonzammi** [reply]
You can get the title and description (or any meta config) in `pageContext.config`. E.g.
```ts
const { title, description } = pageContext.config
```

**Reply by orangecoloured** [reply]
Perfect. Thank you!
Interesting, why do they use `documentProps` in the `preact` preset code for this purpose?
I understand that it's a question that I should probably ask in the preact repo, but maybe it's because of some legacy stuff that used to be in `vike`?

**Reply by phonzammi** [reply]
Let @brillout or @AurelienLourot answers that 🙏

**Reply by lourot** [reply]
This example is incomplete/misleading indeed. I have created https://github.com/brillout/vike-preact/issues/1

**Reply by lourot** [reply]
By the way @orangecoloured if Vike works for you or your company, consider [sponsoring](https://github.com/sponsors/vikejs/) :) It makes a big difference

**Reply by orangecoloured** [reply]
@AurelienLourot Thank you! Just started exploring it. If I end up successfully transitioning my website to preact+vike, I will definitely consider sponsoring. 👌
